Removes abductor ability to use guns.

Shadowlight213 opened the pull request:
Abductors stealing the armory and then running around using tasers to abduct people was always terrible. Hopefully this will reduce that and make them more focused on stealth.
Thanks to coiax's help, Ayys retain the ability to use their gun. Also some abductor check code cleanup.

:cl: Shadowlight and coiax
rscdel: Due to numerous reports of teams using human weapons instead of stealth when carrying out their assignments, future teams have been genetically modified to only be able to use their assigned blaster and are unable to use human weapons.
:/cl:

Shadowlight213 committed:
Removes abductor ability to use guns.
Removes agent starting out with an alien blaster

RemieRichards commented on code/game/gamemodes/miniantags/abduction/abduction.dm:
```diff
@@ -237,7 +237,7 @@
 	agent.equip_to_slot_or_del(V, slot_wear_suit)
 	agent.equip_to_slot_or_del(new /obj/item/weapon/abductor_baton(agent), slot_in_backpack)
-	agent.equip_to_slot_or_del(new /obj/item/weapon/gun/energy/alien(agent), slot_belt)
+	//agent.equip_to_slot_or_del(new /obj/item/weapon/gun/energy/alien(agent), slot_belt)
```
Don't comment out unwanted code, remove it.

Shadowlight213 committed:
removes comment

Cheridan commented:
if all antag strategies are turning into "loot armory", or "destroy armory" in the case of Blob/Nuke... Maybe it's not the antags that are the problem?

Shadowlight213 commented:
@Cheridan remove armory?
Interesting way of thinking.

Robustin commented:
Good change, they're already impossible to catch if they are playing intelligently and the scientist doesn't go AFK. Having them hold more stuns than the rest of sec combined was just stupid since there was no way sec could prevent it (same reason why morph is a shit).

Shadowlight213 commented:
Morph can't use said stuns though. Ayys could.

pudl commented:
@Cheridan it's almost like it was a good idea to put the armory into lockers

Fox-McCloud commented:
They're still going to tele into the armory every time and wipe it out--if only to reduce the availability of weapons the crew has and to nab armor for the scientist.

Shapsy commented:
Still better than letting them just gun everyone down

Anonmare commented:
Could you add basic stun batons to the abductor's lockers then? 
It's gonna be piece of piss annoying for a scientist if they're Agent is an idiot and dies thus forcing you to abduct and dissect all by yourself.

Zilenan91 commented:
Admins should just server ban people who murderbone as ayys. I literally can't think of anything that's shittier.

Jalleo commented:
That or make them exclusively only able to fire the abductor firing pin weapons. Which is the better solution.

coiax commented:
I'm not sure if it's possible to make a firing pin override NOGUN, that would be the trigger guard instead. But ALLOW_ALL trigger guards are rare enough as it is, so making the abductor radiation gun ayypinned and ALLOW_ALL and giving them NOGUN should fix it up.

Also, +1 to the general idea, an ayy team not suffering from lag is almost unstoppable because of recalls, and if they were permitted to murderbone they'd be far far too good at it.

AnturK commented:
I'm not going to argue the need for the change but consider something less artificial than just forbiding all guns. For example emping the teleporting person items or making abductors awful shooters (would need some additional work)

coiax commented:
EMPing anyone who goes through teleportation could work actually.

Shadowlight213 commented:
Ash walkers can't use guns and it works just fine for them.
EMPing the teleportee seems like it would cause a lot more issues than its worth. The agent would have to reactivate their headset on every teleport, it's inconsistent with all other teleporting systems we have, etc..

PKPenguin321 commented:
Make alien gear EMP proof then

Fox-McCloud commented:
@Shadowlight213 

It'll also super-murder people with lots of robotic limbs.

Zilenan91 commented:
We shouldn't balance around EMPs because of how rare they are

PKPenguin321 commented:
That doesn't really apply here zile

Zilenan91 commented:
It does to the suggestions up above

PKPenguin321 commented:
EMPs would not be rare in this context if they were guaranteed though

coiax committed:
Makes Abductors able to use alien pistol

Alien pistol now has ALLOW_ALL trigger guard to overcome abductor's
natural NOGUN.

Pistol now starts in the backpack, and the wondertool on the belt.

In addition:
 - Replace multiple copies of IsAbductor() with a single isabductor()
   proc in code/__HELPERS/mobs.dm
 - Adds icon for alien firing pin
 - Adds abductor span for ayylien communication

Shadowlight213 committed:
Merge pull request #1 from coiax/ayygun-pins

Makes Abductors able to use alien pistol

Anonmare commented:
@PKPenguin321 Abductor headsets are Syndicate Headsets. It'd be a slight buff to Nuke OPs to have EMP-proof headsets unless we snowflake up some unique headsets for them.

Razharas commented on code/__HELPERS/mobs.dm:
```diff
@@ -317,4 +317,11 @@ Proc for attack log creation, because really why not
+	if(ishuman(A))
+		var/mob/living/carbon/human/H = A
+		if(H.dna && istype(H.dna.species, /datum/species/abductor))
+			. = TRUE
```
Why not make this proc atom and override on humans with checks for species type?

coiax replied:
well I wanted to make a macro like the ones in code/__DEFINES/is_helpers.dm but since it requires some variable assignment to get the .dna I couldn't.

Would putting it in atom empty and then putting this body in /mob/living/carbon/human.dm be better? I thought we were trying to avoid empty procs in atom...

PKPenguin321 commented:
>It'd be a slight buff to Nuke OPs to have EMP-proof headsets unless we snowflake up some unique headsets for them.

This is extremely easy
Just make a child path that returns on emp_act

coiax commented:
Nah, NOGUN is better.

SethLafuente commented:
I am from the future and I'm happy to tell you that we found a better solution we simply removed tasers and this came back again when we made batons not be able to be used by chunky fingers a thing which abductors had and we fixed by removing abductor batons from that baton list
